src/manicure.rs

```diff
@@ -79,6 +79,37 @@ pub struct JSErr {
 
 const SOURCE: &str = "nail-salon";
 
+#[wasm_bindgen]
+pub struct Dimensions {
+    pub width: u32,
+    pub height: u32,
+}
+
+#[wasm_bindgen]
+pub fn dimensions(input: &[u8]) -> Result<Dimensions, JsValue> {
+    match image::load_from_memory(&input) {
+        Ok(img) => Ok(Dimensions {
+            width: img.width(),
+            height: img.height(),
+        }),
+        Err(e) => Err(MultiErr::from(e).into())
+    }
+}
+
+#[wasm_bindgen]
+pub struct ScaleAndOrientResult {
+    thumbnail: Vec<u8>,
+    pub width: u32,
+    pub height: u32,
+}
```

Comment:
This does seem to be a convenient interface, though it generates a class with a `free()` method which I believe you need to call manually to avoid a memory leak.